### PR TITLE
Update radar-auth

### PR DESCRIPTION
**Pull request by blootsvoets**
- Use coroutines fetch
- Use radar-auth 2.1.0
- Use radar-commons-kotlin 1.1.1
- Use ktor with kotlinx-serialization
- Use `buildSrc` and `radar-commons-gradle` for building
- Use async for serving responses. Especially relevant for instances with a lot of users in the connector.

### CODE REVIEW

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OuraAuthorizationService.kt [hunk 1]
```diff
@@ -27,28 +23,34 @@ class OuraAuthorizationService(
...
-
-        var requestObj = Request.Builder().apply {
-            url(revokeURI)
-            post(FormBody.Builder().build())
-            header("Authorization", credentials)
-            header("Content-Type", "application/x-www-form-urlencoded")
-            header("Accept", "application/json")
-        }.build()
+        val isSuccess = try {
+            val response = httpClient.submitForm {
```
Comment (mpgxvii) [inline authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OuraAuthorizationService.kt:27]: I haven't really used `withContext(Dispatchers.IO)` much, but should it be used here as well?

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt [hunk 2]
```diff
@@ -79,111 +94,112 @@ abstract class OAuth1RestSourceAuthorizationService(
...
         params[OAUTH_SIGNATURE] =
             OauthSignature(url, params, method, authConfig.clientSecret, tokens.tokenSecret).getEncodedSignature()
-        val headers = params.toFormattedHeader()
 
-        return Request.Builder()
-            .url(url)
-            .header("Authorization", "OAuth $headers")
-            .method(method, if (method == "POST") "".toRequestBody(null) else null)
-            .build()
+        return httpClient.request(url = Url(url)) {
```
Comment (mpgxvii) [inline authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt:194]: If we run `withContext` here, and remove it from the methods that call `request()`, will those still run in the context? Or would we need to still run `withContext` in the outer methods?

**Review by mpgxvii — DISMISSED** [review]
LGTM, thanks!

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OuraAuthorizationService.kt (hunk 1 above)
Comment (blootsvoets) [reply authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OuraAuthorizationService.kt]: Good find, yes that's right.

File: authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt (hunk 2 above)
Comment (blootsvoets) [reply authorizer-app-backend/src/main/java/org/radarbase/authorizer/service/OAuth1RestSourceAuthorizationService.kt]: yes, you're right, I've added it to the lowermost function now. Indeed `withContext` calls stack without problem. If it's already in the right context it won't do any switches.

**Pull request merged**